torch/utils/data/datapipes/iter/grouping.py

```diff
@@ -28,6 +28,12 @@ def __iter__(self):
             if i % self.num_of_instances == self.instance_id:
                 yield item
 
+    def __len__(self):
+        if isinstance(self.source_datapipe, Sized):
+            return len(self.source_datapipe) // self.num_of_instances +\
+                (1 if (self.instance_id < len(self.source_datapipe) % self.num_of_instances) else 0)
```

Comment:
I guess it's nit: you can do `(len(self.source_datapipe) + self.num_of_instances -1) // self.num_of_instances` to get the same result without ternary op.

Reply:
I thought about it but was worried that it was unintuitive for readers. I will switch to that then.

Reply:
Actually, it needs to be
`(len(self.source_datapipe) + (self.num_of_instances - self.instance_id) - 1) // self.num_of_instances`

So I think it might be better to leave it as it was.